In [ ]:
import pandas as pd
import numpy as np

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
!ls "/content/drive/My Drive/Research_Notebooks/Thomas-Prototype/"

thomas_fradulent_emails.txt


In [ ]:
file_path = "/content/drive/My Drive/Research_Notebooks/Thomas-Prototype/thomas_fradulent_emails.txt"

try:
    with open(file_path, "r", encoding="utf-8", errors = 'ignore') as file:
        emails_contents = file.read()
except Exception as e:
    print(f"An error occurred: {e}")


In [ ]:
emails = emails_contents.split("From r")[1:]

df_all = pd.DataFrame(emails,columns=['email_content'])

In [ ]:
print(df_all.iloc[1]['email_content'])

  Thu Oct 31 08:11:39 2002
Return-Path: <bensul2004nng@spinfinder.com>
X-Sieve: cmu-sieve 2.0
Return-Path: <bensul2004nng@spinfinder.com>
Message-Id: <200210311310.g9VDANt24674@bloodwork.mr.itd.UM>
From: "Mr. Ben Suleman" <bensul2004nng@spinfinder.com>
Date: Thu, 31 Oct 2002 05:10:00
To: R@M
Subject: URGENT ASSISTANCE /RELATIONSHIP (P)
MIME-Version: 1.0
Content-Type: text/plain;charset="iso-8859-1"
Content-Transfer-Encoding: 7bit
Status: O

Dear Friend,

I am Mr. Ben Suleman a custom officer and work as Assistant controller of the Customs and Excise department Of the Federal Ministry of Internal Affairs stationed at the Murtala Mohammed International Airport, Ikeja, Lagos-Nigeria.

After the sudden death of the former Head of state of Nigeria General Sanni Abacha on June 8th 1998 his aides and immediate members of his family were arrested while trying to escape from Nigeria in a Chartered jet to Saudi Arabia with 6 trunk boxes Marked "Diplomatic Baggage". Acting on a tip-off as they at

In [ ]:
sample_email = df_all.iloc[1]['email_content']

'  Thu Oct 31 08:11:39 2002\nReturn-Path: <bensul2004nng@spinfinder.com>\nX-Sieve: cmu-sieve 2.0\nReturn-Path: <bensul2004nng@spinfinder.com>\nMessage-Id: <200210311310.g9VDANt24674@bloodwork.mr.itd.UM>\nFrom: "Mr. Ben Suleman" <bensul2004nng@spinfinder.com>\nDate: Thu, 31 Oct 2002 05:10:00\nTo: R@M\nSubject: URGENT ASSISTANCE /RELATIONSHIP (P)\nMIME-Version: 1.0\nContent-Type: text/plain;charset="iso-8859-1"\nContent-Transfer-Encoding: 7bit\nStatus: O\n\nDear Friend,\n\nI am Mr. Ben Suleman a custom officer and work as Assistant controller of the Customs and Excise department Of the Federal Ministry of Internal Affairs stationed at the Murtala Mohammed International Airport, Ikeja, Lagos-Nigeria.\n\nAfter the sudden death of the former Head of state of Nigeria General Sanni Abacha on June 8th 1998 his aides and immediate members of his family were arrested while trying to escape from Nigeria in a Chartered jet to Saudi Arabia with 6 trunk boxes Marked "Diplomatic Baggage". Acting on a

In [ ]:
# Split the text into header and body sections
header, body = sample_email.split("Dear", 1)

# Remove any leading or trailing whitespace from the header and body sections
header = header.strip()
body = body.strip()

print("Header Section:")
print(header)
print("\nBody Section:")
print(body)


Header Section:
Thu Oct 31 08:11:39 2002
Return-Path: <bensul2004nng@spinfinder.com>
X-Sieve: cmu-sieve 2.0
Return-Path: <bensul2004nng@spinfinder.com>
Message-Id: <200210311310.g9VDANt24674@bloodwork.mr.itd.UM>
From: "Mr. Ben Suleman" <bensul2004nng@spinfinder.com>
Date: Thu, 31 Oct 2002 05:10:00
To: R@M
Subject: URGENT ASSISTANCE /RELATIONSHIP (P)
MIME-Version: 1.0
Content-Type: text/plain;charset="iso-8859-1"
Content-Transfer-Encoding: 7bit
Status: O

Body Section:
Friend,

I am Mr. Ben Suleman a custom officer and work as Assistant controller of the Customs and Excise department Of the Federal Ministry of Internal Affairs stationed at the Murtala Mohammed International Airport, Ikeja, Lagos-Nigeria.

After the sudden death of the former Head of state of Nigeria General Sanni Abacha on June 8th 1998 his aides and immediate members of his family were arrested while trying to escape from Nigeria in a Chartered jet to Saudi Arabia with 6 trunk boxes Marked "Diplomatic Baggage". Acting 

In [ ]:
print(body)

Friend,

I am Mr. Ben Suleman a custom officer and work as Assistant controller of the Customs and Excise department Of the Federal Ministry of Internal Affairs stationed at the Murtala Mohammed International Airport, Ikeja, Lagos-Nigeria.

After the sudden death of the former Head of state of Nigeria General Sanni Abacha on June 8th 1998 his aides and immediate members of his family were arrested while trying to escape from Nigeria in a Chartered jet to Saudi Arabia with 6 trunk boxes Marked "Diplomatic Baggage". Acting on a tip-off as they attempted to board the Air Craft,my officials carried out a thorough search on the air craft and discovered that the 6 trunk boxes contained foreign currencies amounting to US$197,570,000.00(One Hundred and  Ninety-Seven Million Five Hundred Seventy Thousand United States Dollars).

I declared only (5) five boxes to the government and withheld one (1) in my custody containing the sum of (US$30,000,000.00) Thirty Million United States Dollars Only, 

In [ ]:
from nltk.tokenize import word_tokenize

# Tokenize the text
word_tokens = word_tokenize(body)

# Count the number of words
num_words = len(word_tokens)

print("Body Size:", num_words)


Body Size: 592


In [ ]:
from nltk.corpus import stopwords
import pprint


stop_words = set(stopwords.words('english'))

# Filter out stopwords
filtered_words = [w for w in word_tokens if not w.lower() in stop_words]

# Join the filtered words into a single string
filtered_body = ' '.join(filtered_words)

# Word count of original body text
original_word_count = len(word_tokens)
print("Original word count:", original_word_count)

filtered_word_count = len(filtered_words)
print("Filtered word count:", filtered_word_count)

# Pretty print the filtered text for readability
pprint.pprint(filtered_body)



Original word count: 592
Filtered word count: 341
('Friend , Mr. Ben Suleman custom officer work Assistant controller Customs '
 'Excise department Federal Ministry Internal Affairs stationed Murtala '
 'Mohammed International Airport , Ikeja , Lagos-Nigeria . sudden death former '
 'Head state Nigeria General Sanni Abacha June 8th 1998 aides immediate '
 'members family arrested trying escape Nigeria Chartered jet Saudi Arabia 6 '
 "trunk boxes Marked `` Diplomatic Baggage '' . Acting tip-off attempted board "
 'Air Craft , officials carried thorough search air craft discovered 6 trunk '
 'boxes contained foreign currencies amounting US $ 197,570,000.00 ( One '
 'Hundred Ninety-Seven Million Five Hundred Seventy Thousand United States '
 'Dollars ) . declared ( 5 ) five boxes government withheld one ( 1 ) custody '
 'containing sum ( US $ 30,000,000.00 ) Thirty Million United States Dollars , '
 'disguised prevent discovered transportation process.Due several media '
 'reports late he

In [ ]:
percentage_reduction = (filtered_word_count  / original_word_count) * 100
print(percentage_reduction)

57.60135135135135


After running stopwords, the body content of the email was reduced to **57.6%** of the original size.

After reading the filtered email, the content is still comprehensible and successfully reduced.

Exploration of other extraction tools are necessary for improved summarisation.